docs/nx-cloud/concepts/cache-security.md

```diff
@@ -0,0 +1,59 @@
+# Remote Cache Security
+
+{% callout type="warning" title="Use Caution With Read-Write Tokens" %}
+Read-write tokens allow full write access to your remote cache. They should only be used in trusted environments. For instance, open source projects should only use read-write tokens as secrets configured for protected branches (e.g, main). Read-only tokens should be used in all other cases.
+{% /callout %}
+
+A cache allows you to reuse work that has already been done, but it also introduces a potential security risk - cache poisoning. A poisoned cache is one where the cached files have been altered in some way by a malicious actor. When a developer or the CI pipeline use that poisoned cache, the task output will be what the malicious actor wants instead of the correct task output.
+
+Nx takes security seriously and has put in place many precautions (we're [SOC 2 compliant](https://security.nx.app)). Listed below are some precautions that you need to take in your own codebase.
+
+## Recommended Precautions
+
+In order to keep your cache secure, there are a few steps we recommend you take:
+
+- Specify a [read-only token](/nx-cloud/recipes/security/access-tokens) in the `nx.json` file
+- Use a [read-write token](/nx-cloud/recipes/security/access-tokens) only in the `main` branch in CI, not on pull request branches.
+- When building the actual deployment artifact, [skip the cache](/core-features/cache-task-results#turn-off-or-skip-the-cache)
+- Do not [manually share your local cache](/recipes/troubleshooting/unknown-local-cache) folder. Use Nx Cloud [remote caching](/nx-cloud/features/remote-cache) instead.
+- Configure [end to end encryption](/nx-cloud/recipes/security/encryption)
+
+If you need to have all cache artifacts on servers that you control, there is an on-premise version of Nx Cloud that you can use as part of the [Enterprise plan](https://nx.app/enterprise).
+
+## Consequences of a Security Incident
+
+Nx does not send your actual code to the remote cache. There are 3 kinds of data that are sent to the Nx Cloud remote cache for each task:
+
+1. A hash of the inputs to the task. There is no way to reconstitute the actual source code files that were used to create a particular hash value.
+2. Any files that were created as outputs from a task.
+3. The terminal output created when running the task.
+
+If a malicious actor were able to modify the cache and those output files were then executed, that malicious actor could run arbitrary code on developer machines or in CI.
+
+## Potential Malicious Actors
+
+The following section describes different personas that might be malicious actors and how the precautions taken above would counter their nefarious intent.
```

Comment:
Just came across this now, so this is actually the explainer for the recommendations. Maybe we can switch the sections, like move the "Consequences of a Security Incident" first but renaming it to something like "What data is stored".

I'd like to see a more clearer connection though between the recommendation, the why and what it can prevent